parachains/runtimes/collectives/collectives-polkadot/src/ambassador/mod.rs

```diff
@@ -0,0 +1,141 @@
+//! - Managed content (charter, announcements) (via [pallet_collective_content]).
+
+pub mod origins;
+mod tracks;
+
+use super::*;
+use frame_support::traits::TryMapSuccess;
+pub use origins::pallet_origins as pallet_ambassador_origins;
+use origins::pallet_origins::{
+	EnsureAmbassador, EnsureRankedAmbassador, EnsureSeniorAmbassador, Origin,
+};
+use sp_arithmetic::traits::CheckedSub;
+use sp_runtime::{
+	morph_types,
+	traits::{ConstU16, TypedGet},
+};
+
+/// The Ambassador Program's member ranks.
+pub mod ranks {
+	use pallet_ranked_collective::Rank;
+
+	#[allow(dead_code)]
+	pub const CANDIDATE: Rank = 0;
+	pub const AMBASSADOR: Rank = 1;
+	pub const SENIOR_AMBASSADOR: Rank = 2;
+	pub const HEAD_AMBASSADOR: Rank = 3;
+}
+
+pub type AmbassadorContentInstance = pallet_collective_content::Instance1;
+
+impl pallet_collective_content::Config<AmbassadorContentInstance> for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type CharterOrigin = EnsureAmbassador;
+	// An announcement can be submitted by a Senior Ambassador member or the [Origin::Ambassador] plurality voice
+	// taken via referendum on the [tracks::constants::AMBASSADOR] track.
+	type AnnouncementOrigin = EitherOfDiverse<
+		pallet_ranked_collective::EnsureMember<
+			Runtime,
+			AmbassadorCollectiveInstance,
+			{ ranks::SENIOR_AMBASSADOR },
+		>,
+		EnsureAmbassador,
+	>;
+	type WeightInfo = weights::pallet_collective_content::WeightInfo<Runtime>;
+}
+
+impl pallet_ambassador_origins::Config for Runtime {}
+
+parameter_types! {
+	pub const AlarmInterval: BlockNumber = 1;
+	pub const SubmissionDeposit: Balance = 0;
+	pub const UndecidingTimeout: BlockNumber = 7 * DAYS;
+}
+
+pub type AmbassadorReferendaInstance = pallet_referenda::Instance1;
+
+impl pallet_referenda::Config<AmbassadorReferendaInstance> for Runtime {
+	type WeightInfo = weights::pallet_referenda::WeightInfo<Runtime>;
+	type RuntimeCall = RuntimeCall;
+	type RuntimeEvent = RuntimeEvent;
+	type Scheduler = Scheduler;
+	type Currency = Balances;
+	// A proposal can be submitted by a member of the Ambassador Program of [ranks::AMBASSADOR] rank or higher.
+	type SubmitOrigin = pallet_ranked_collective::EnsureMember<
+		Runtime,
+		AmbassadorCollectiveInstance,
+		{ ranks::AMBASSADOR },
+	>;
+	type CancelOrigin = EitherOf<EnsureRoot<AccountId>, EnsureSeniorAmbassador>;
+	type KillOrigin = EitherOf<EnsureRoot<AccountId>, EnsureSeniorAmbassador>;
+	type Slash = ToParentTreasury<Runtime>;
+	type Votes = pallet_ranked_collective::Votes;
+	type Tally = pallet_ranked_collective::TallyOf<Runtime, AmbassadorCollectiveInstance>;
+	type SubmissionDeposit = SubmissionDeposit;
+	type MaxQueued = ConstU32<20>;
+	type UndecidingTimeout = UndecidingTimeout;
+	type AlarmInterval = AlarmInterval;
+	type Tracks = tracks::TracksInfo;
+	type Preimages = Preimage;
+}
+
+parameter_types! {
+	// TODO remove "+10", use [ranks::HEAD_AMBASSADOR] instead. To make this possible the benches
```

Comment:
Should this todo be fixed?

Reply:
Removed the todo. 
To remove the needs for the max rank to be at least 10, the new api configuration will need to be introduced.
I think it does not worst it.